rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -418,7 +418,26 @@ void main_window::InstallPup()
 		updatefilenames.begin(), updatefilenames.end(), [](std::string s) { return s.find("dev_flash_") == std::string::npos; }),
 		updatefilenames.end());
 
-	QProgressDialog pdlg(tr("Installing firmware ... please wait ..."), tr("Cancel"), 0, static_cast<int>(updatefilenames.size()), this);
+	fs::file version_file_f = pup.get_file(0x100);
+	std::string version_string;
+	version_string.resize(version_file_f.size());
+	version_file_f.read(version_string);
+	version_string.erase(version_string.find('\n'));
+
+	const int cur_major = 4;
+	const int cur_minor = 81;
+
+	int major = std::stoi(version_string.substr(0, version_string.find('.')));
+	int minor = std::stoi(version_string.substr(version_string.find('.') + 1));
+
+	if ((major < cur_major || (major == cur_major && minor < cur_minor)) &&
```

Comment:
you could just use qstr(version_string).toDouble();
or do you explicitly need both major and minor?
this would save you a lot of code and you could use it for the messagebox as well

btw. wouldn't it be better to put your whole code up in front of that other part in line 413?